### ISSUE TITLE
Comparing only fields can pass if you mistype the fieldname

### ISSUE THREAD

**Issue by gibamizu** [assertj/assertj#3270]
**Describe the bug**
When comparing using ```comparingOnlyFields``` with mistyped field name, the assertion pass.

* assertj core version: 3.24.2
* java version: 17.0.8.1

**Example**

```java
public class AssertJTest implements WithAssertions {
    static class A1 {
        public final int a, b;
        public A1(int a, int b) {
            this.a = a;
            this.b = b;
        }
    }

    @Test
    public void testComparingOnly() {
        A1 a1 = new A1(1, 2);
        A1 a2 = new A1(3, 4);

        // This passes if we specify a non-existent field
        assertThat(a1).usingRecursiveComparison()
                .comparingOnlyFields("xxx")
                .isEqualTo(a2);
    }
}

```

**Comment by robinmatz** [comment]
Hey @gibamizu ,

this issue does not reproduce with the current development branch, where it fails with the following exception
```
java.lang.IllegalArgumentException: The following fields don't exist: {xxx}

	at org.assertj.core.api.recursive.comparison.RecursiveComparisonConfiguration.checkComparedFieldsExist(RecursiveComparisonConfiguration.java:1086)
	at org.assertj.core.api.recursive.comparison.RecursiveComparisonDifferenceCalculator$ComparisonState.initDualValuesToCompare(RecursiveComparisonDifferenceCalculator.java:142)
	at org.assertj.core.api.recursive.comparison.RecursiveComparisonDifferenceCalculator$ComparisonState.access$100(RecursiveComparisonDifferenceCalculator.java:73)
	at org.assertj.core.api.recursive.comparison.RecursiveComparisonDifferenceCalculator.determineDifferences(RecursiveComparisonDifferenceCalculator.java:233)
	at org.assertj.core.api.recursive.comparison.RecursiveComparisonDifferenceCalculator.determineDifferences(RecursiveComparisonDifferenceCalculator.java:224)
	at org.assertj.core.api.RecursiveComparisonAssert.determineDifferencesWith(RecursiveComparisonAssert.java:1770)
	at org.assertj.core.api.RecursiveComparisonAssert.isEqualTo(RecursiveComparisonAssert.java:157)
	at org.assertj.core.api.recursive.comparison.RecursiveComparisonAssert_isEqualTo_comparingOnlyFields_Test.should_also_fail_when_field_does_not_exist(RecursiveComparisonAssert_isEqualTo_comparingOnlyFields_Test.java:423)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
```

Seemingly, this has been fixed with #3129, as there is a test for it already tagged with this issue, and will be released in the next version.

**Comment by joel-costigliola** [comment]
Indeed @robinmatz thanks for looking into that.
